server/email_verification.go

```diff
@@ -191,52 +216,128 @@ type emailVerifiedTemplateData struct {
 	Message string
 }
 
-func handleEmailVerifyFunc(verifiedTpl *template.Template, issuer url.URL, keysFunc func() ([]key.PublicKey,
+func handleEmailVerifyFunc(verifiedTpl *template.Template, s *sessionmanager.SessionManager, issuer url.URL, keysFunc func() ([]key.PublicKey,
 	error), userManager *manager.UserManager) http.HandlerFunc {
 
 	return func(w http.ResponseWriter, r *http.Request) {
 		q := r.URL.Query()
 		token := q.Get("token")
 
+		runtpl := func(error string, message string, statusCode int) {
+			execTemplateWithStatus(w, verifiedTpl, emailVerifiedTemplateData{
+				Error:   error,
+				Message: message,
+			}, statusCode)
+		}
+
 		keys, err := keysFunc()
 		if err != nil {
-			execTemplateWithStatus(w, verifiedTpl, emailVerifiedTemplateData{
-				Error:   "There's been an error processing your request.",
-				Message: "Please try again later.",
-			}, http.StatusInternalServerError)
+			runtpl("There's been an error processing your request.", "Please try again later.", http.StatusInternalServerError)
 			return
 		}
 
 		ev, err := user.ParseAndVerifyEmailVerificationToken(token, issuer, keys)
 		if err != nil {
-			execTemplateWithStatus(w, verifiedTpl, emailVerifiedTemplateData{
-				Error:   "Bad Email Verification Token",
-				Message: "That was not a verifiable token.",
-			}, http.StatusBadRequest)
+			runtpl("Bad Email Verification Token", "That was not a verifiable token.", http.StatusBadRequest)
+			return
+		}
+
+		aud, ok, err := ev.Claims.StringClaim("aud")
```

Comment:
using a helperFunc you'll have a problem: `return` statement will exit helper func, not the main func.

alternative: I tried using `goto` but you cannot reference a label outside the helperFunc, so I ran out of alternatives :P